build_tools/cmake/iree_macros.cmake

```diff
@@ -514,6 +514,37 @@ function(iree_symlink_tool)
   )
 endfunction()
 
+#-------------------------------------------------------------------------------
+# Test timeout values
+#-------------------------------------------------------------------------------
+
+# iree_get_timeout_seconds()
+#
+# Evaluates a Bazel-like timeout string to a number of seconds.
+# Allowed timeout values:  "short", "moderate", "long".
+# Default is "short".
+#
+# This is aligned with our `native_test` function in Bazel. This differs from
+# general Bazel rules in two ways: they default to "moderate" and they allow an
+# additional timeout value "eternal".
```

Comment:
It seems like it would be simpler to support "eternal" than to explain everywhere that we don't support it. I guess this is similar to my suggestion that we just set `timeout = "short"` as the default arg to all our test macros. That's not redefining any Bazel behavior and we don't need to explain it: it's just that one of the convenience behaviors of this macro is to default to "short" (which honestly should've been the default to begin with IMO).